Mage.Sets/src/mage/cards/f/FranticScapegoat.java

```diff
@@ -0,0 +1,166 @@
+package mage.cards.f;
+
+import mage.MageInt;
+import mage.MageItem;
+import mage.MageObjectReference;
+import mage.abilities.Ability;
+import mage.abilities.TriggeredAbilityImpl;
+import mage.abilities.common.EntersBattlefieldTriggeredAbility;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.SuspectSourceEffect;
+import mage.abilities.keyword.HasteAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+import mage.constants.SubType;
+import mage.constants.Zone;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.filter.predicate.permanent.PermanentReferenceInCollectionPredicate;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.events.ZoneChangeGroupEvent;
+import mage.game.permanent.Permanent;
+import mage.players.Player;
+import mage.target.Target;
+import mage.target.TargetPermanent;
+
+import java.util.Objects;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ *
+ * @author notgreat
+ */
+public final class FranticScapegoat extends CardImpl {
+
+    public FranticScapegoat(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{R}");
+        
+        this.subtype.add(SubType.GOAT);
+        this.power = new MageInt(1);
+        this.toughness = new MageInt(1);
+
+        // Haste
+        this.addAbility(HasteAbility.getInstance());
+
+        // When Frantic Scapegoat enters the battlefield, suspect it.
+        this.addAbility(new EntersBattlefieldTriggeredAbility(new SuspectSourceEffect()));
+
+        // Whenever one or more other creatures enter the battlefield under your control, if Frantic Scapegoat is suspected, you may suspect one of the other creatures. If you do, Frantic Scapegoat is no longer suspected.
+        this.addAbility(new FranticScapegoatTriggeredAbility());
+
+    }
+
+    private FranticScapegoat(final FranticScapegoat card) {
+        super(card);
+    }
+
+    @Override
+    public FranticScapegoat copy() {
+        return new FranticScapegoat(this);
+    }
+}
+
+//Based on Ingenious Artillerist and Lightmine Field
+class FranticScapegoatTriggeredAbility extends TriggeredAbilityImpl {
+
+    FranticScapegoatTriggeredAbility() {
+        super(Zone.BATTLEFIELD, new FranticScapegoatSuspectEffect(), true);
+    }
+
+    private FranticScapegoatTriggeredAbility(final FranticScapegoatTriggeredAbility ability) {
+        super(ability);
+    }
+
+    @Override
+    public boolean checkEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.ZONE_CHANGE_GROUP;
+    }
+
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        Permanent source = getSourcePermanentIfItStillExists(game);
+        if (source == null || !source.isSuspected()){
```

Comment:
style, can you please tweak your IDE to leave space between closing parenthesis and opening brace

Reply:
I don't use IntelliJ for anything other than this, as far as I can tell that's already configured. Do I need to turn on strict code style enforcement somehow?

Reply:
Some hints:
* If you use IntelliJ IDE then use auto-format for big changes or newly added files. It already configured by default — just press CTRL+L to format a file (for new files I recommends to use CTRL+I to optimize imports section too).
* Warning, don’t use auto-format on small changes or massive refactor (many files changes);
* If you use other IDE like NetBeans or VSC then don’t use auto-format (it can be incompatible with IntelliJ and generates too much changes).